Stripe/STPAPIClient.m

```diff
@@ -115,33 +113,34 @@ + (instancetype)sharedClient {
 }
 
 - (instancetype)init {
-    return [self initWithConfiguration:[STPPaymentConfiguration sharedConfiguration]];
-}
-
-- (instancetype)initWithPublishableKey:(NSString *)publishableKey {
-    STPPaymentConfiguration *config = [[STPPaymentConfiguration alloc] init];
-    config.publishableKey = [publishableKey copy];
-    return [self initWithConfiguration:config];
-}
-
-- (instancetype)initWithConfiguration:(STPPaymentConfiguration *)configuration {
-    NSString *publishableKey = [configuration.publishableKey copy];
-    if (publishableKey) {
-        [self.class validateKey:publishableKey];
-    }
     self = [super init];
     if (self) {
-        _apiKey = publishableKey;
         _apiURL = [NSURL URLWithString:APIBaseURL];
-        _configuration = configuration;
-        _stripeAccount = configuration.stripeAccount;
+        _configuration = [STPPaymentConfiguration sharedConfiguration];
         _sourcePollers = [NSMutableDictionary dictionary];
         _sourcePollersQueue = dispatch_queue_create("com.stripe.sourcepollers", DISPATCH_QUEUE_SERIAL);
         _urlSession = [NSURLSession sessionWithConfiguration:[self.class sharedUrlSessionConfiguration]];
     }
     return self;
 }
 
+- (instancetype)initWithPublishableKey:(NSString *)publishableKey {
+    STPAPIClient *apiClient = [self init];
+    apiClient.publishableKey = publishableKey;
+    return apiClient;
+}
+
+- (instancetype)initWithConfiguration:(STPPaymentConfiguration *)configuration {
```

Comment:
since all the behavior here is deprecated, should this initializer be deprecated as well?

Reply:
I think that's fair.  I wouldn't expect many users of this, anyways.